app/controllers/ApplicationController.scala

```diff
@@ -80,16 +82,39 @@ class PdfDownloadController @Inject()(appConfig: FrontendAppConfig,
     }
   }
 
-  def ruling(reference: String, token: Option[String]): Action[AnyContent] = identify.async { implicit request =>
+  def rulingCertificatePdf(reference: String, token: Option[String]): Action[AnyContent] = identify.async { implicit request =>
     getPdf(request.eoriNumber, reference, token, getRulingPDF)
   }
 
   private def getRulingPDF(eori: Eori, reference: CaseReference)
                           (implicit request: Request[AnyContent]): Future[Result] = {
     caseService.getCaseWithRulingForUser(eori, reference) flatMap { c: Case =>
       lazy val decision = c.decision.getOrElse(throw new IllegalStateException("Missing decision"))
-      generatePdf(rulingPdf(appConfig, c, decision), s"BTIRuling$reference.pdf")
+      generatePdf(rulingCertificateTemplate(appConfig, c, decision), s"BTIRuling$reference.pdf")
+    }
+  }
+
+  def viewRulingCertificate(reference: String, token: Option[String]): Action[AnyContent] = identify.async { implicit request =>
+    request.eoriNumber match {
+      case Some(eori: String) =>
+        caseService.getCaseWithRulingForUser(eori, reference) flatMap {
+          c: Case => successful(Ok(rulingCertificateView(appConfig, c)))
+        }
+      case None => successful(Redirect(routes.BeforeYouStartController.onPageLoad()))
     }
   }
 
+  def viewApplication(reference: String, token: Option[String]): Action[AnyContent] = identify.async { implicit request =>
+    request.eoriNumber match {
```

Comment:
What if we need to invoke the CDS workaround and users ned access without an EORI?
Thats what the token is for which you appear not to be using

Reply:
This should now be addressed